Cookbook/Proposals/SplitOctopus.md

```diff
@@ -0,0 +1,65 @@
+# Split Octopus codebase
+
+* Author(s): Giorgos Tsiapaliokas
+* Review Managers: David Rodrigues, Yasuhiro Inami, Olivier Halligon
+
+## Introduction
+
+From the early versions of the codebase Octopus was the place were features were added.
+Unfortunately Octopus hasn't adapted as the rest of the codebase has.
+In the iOS codebase we use frameworks in order to organize the code and Octopus isn't one.
+
+## Motivation
+
+While there is still the need for having a place for small features Octopus has become a massive container which contains both features and services.
+Also from a tech POV Octopus is just a folder which is being reused in different targets.
+I am sure we can do better than this :)
+
+## Proposed solution
+
+Split Octopus into two different frameworks.
+The proposed names are `BabylonFeatures` and `BabylonApp`.
+
+The first framework will be used as a container for features which
+- depend only on `BabylonCore`, `BabylonUI`, `BabylonDepedencies` and `BabylonSDK`
+- don't depend on any third party SDKs
```

Comment:
What about first party dependencies, like StoreKit, or other other frameworks? This way most of the staff will still live in the "app" framework where the name of it suggests that it should contain only stuff related to the "app level" code, like app delegate related code and app lifecycle, or things like tab bar and MainUserSession. How depending on StoreKit for instance justifies keeping payments functionality in this "app" framework when it is clearly a "feature" that should be actually consumed by other frameworks, i.e. test and kits should be able to use payments functionality (it does not have to depend on the payments framework for that though, but it could), but test and kits are moving to features but payments stay at the app level. What is the dependency diagram then?

Imo we need to discuss a long term plan for the split, even if we are doing it in stages like first splitting into smaller number of frameworks and them split these frameworks in smaller pieces.